release notes/v0.54.0.md

```diff
@@ -0,0 +1,304 @@
+
+This includes support for numeric literal separators, like:
+
+```javascript
+const billion = 1_000_000_000
+```
+
+Support for `BigInt`, the values which are too large to be represented by the number primitive.
+
+```javascript
+const huge = BigInt(9007199254740991);
+```
+
+Note: that prior k6 version v0.54, Golang's type `math/big.Int` mapped to another type, so it might be a breaking change for some extensions or users.
+
+Regexp dotall support - letting you match newline characters with `.`.
+
+```javascript
+const str1 = "bar\nexample foo example";
+
+const regex1 = /bar.example/s;
+
+console.log(regex1.dotAll); // true
+```
+
+Support for ES2023 Array methods: [`with`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/with), [`toSpliced`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSpliced), [`toReversed`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toReversed) and [`toSorted`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted).
+
+Thank you @shiroyk for adding both the new array methods and BitInt :bow:.
+
+### New `setChecked` method for the browser module [`browser#1403`](https://github.com/grafana/xk6-browser/pull/1403)
+
+Previously, users could check or uncheck checkbox and radio button elements using the [`check`](https://grafana.com/docs/k6/latest/javascript-api/k6-browser/page/check/) and [`uncheck`](https://grafana.com/docs/k6/latest/javascript-api/k6-browser/page/uncheck/) methods. Now, we've added a `setChecked` method that allows users to set a checkbox or radio button to either the checked or unchecked state with a single method and a boolean argument.
+
+```javascript
+await page.setChecked('#checkbox', true);   // check the checkbox
+await page.setChecked('#checkbox', false);  // uncheck the checkbox
+```
+
+[Page](https://grafana.com/docs/k6/latest/javascript-api/k6-browser/page/setchecked/), [Frame](https://grafana.com/docs/k6/latest/javascript-api/k6-browser/frame/setchecked/), [ElementHandle](https://grafana.com/docs/k6/latest/javascript-api/k6-browser/elementhandle//setchecked/), and [Locator](https://grafana.com/docs/k6/latest/javascript-api/k6-browser/locator/setchecked/) now support the new `setChecked` method.
+
+### Async `check` function utility [`k6-utils#13`](https://github.com/grafana/k6-jslib-utils/pull/13)
+
+Writing clean code can be cumbersome using [the k6 `check` function](https://grafana.com/docs/k6/latest/javascript-api/k6/check/) with async code because the `check` function does not support asynchronous checks. One potential solution is to declare a temporary variable and wait for the value first, then check the result later. However, this approach can clutter the code with single-use declarations and unnecessary variable names:
```

Comment:
```suggestion
Writing concise code can be difficult when using [the k6 `check` function](https://grafana.com/docs/k6/latest/javascript-api/k6/check/) with async code since it doesn't support async APIs. A solution that we have suggested so far is to declare a temporary variable, wait for the value that is to be checked, and then check the result later. However, this approach can clutter the code with single-use declarations and unnecessary variable names, e.g.:
```

@inancgumus WDYT?

Reply:
Perfect!